contracts/xrpl-multisig-prover/src/contract/reply.rs

```
@@ -0,0 +1,112 @@
+                CROSS_CHAIN_ID_TO_MULTISIG_SESSION.save(
+                    deps.storage,
+                    cc_id,
+                    &MultisigSession {
+                        id: multisig_session_id.u64(),
+                        expires_at,
+                    },
+                )?;
+                REPLY_CROSS_CHAIN_ID.remove(deps.storage);
+            };
+
+            REPLY_UNSIGNED_TX_HASH.remove(deps.storage);
+
+            Ok(Response::new()
+                .add_event(Event::ProofUnderConstruction {
+                    destination_chain: config.chain_name,
+                    unsigned_tx_hash: unsigned_tx_hash.clone(),
+                    multisig_session_id,
+                })
+                .add_event(
+                    cosmwasm_std::Event::new("xrpl_signing_started")
+                        .add_attributes(
+                            signing_started_attributes
+                                .into_iter()
+                                .filter(|a| !a.key.starts_with('_') && a.key != "msg")
+                                .collect::<Vec<_>>(),
+                        )
+                        .add_attribute(
+                            "unsigned_tx",
+                            HexBinary::from(
+                                XRPLUnsignedTxToSign {
+                                    unsigned_tx: tx_info.unsigned_tx,
+                                    unsigned_tx_hash,
+                                    cc_id: opt_cc_id,
+                                }
+                                .xrpl_serialize()?,
+                            )
+                            .to_hex(),
+                        ),
+                ))
+        }
+        Ok(MsgExecuteContractResponse { data: None }) => Err(ContractError::InvalidContractReply {
+            reason: "no data".to_string(),
+        }),
+        Err(_) => {
+            unreachable!("violated invariant: replied failed submessage with ReplyOn::Success")
```

I believe this Error is referring to `ParseReplyError` enum right? Can we return the proper error type (I believe it has three options) alongside this message?

---

Sounds good, added here: https://github.com/commonprefix/axelar-amplifier/pull/4/commits/015ccf6418cbb7cc04bcf1e93c89fe6abd1a437d